packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts

```diff
@@ -0,0 +1,83 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { StreamEventSource, StreamEventSourceProps } from './stream';
+
+export interface KafkaEventSourceProps extends StreamEventSourceProps {
+}
+
+/**
+ * Use a MSK cluster as a streaming source for AWS Lambda
+ */
+export class ManagedKafkaEventSource extends StreamEventSource {
+  /**
+   * Create an event source for MSK
+   * @param clusterArn - the ARN of the MSK cluster
+   * @param topic - the Kafka topic to subscribe to
+   * @param secret - the secret with the Kafka credentials, see https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html for details
+   * @param props
+   */
+  constructor(
+    readonly clusterArn: string,
+    readonly topic: string,
+    readonly secret: secretsmanager.ISecret,
```

Comment:
Move topic and secret as properties inside `KafkaEventSourceProps`.

As for cluster, there are two options - if you want to keep it outside the props as a separate attribute, I recommend you implement the higher level construct for MSK (the simplest working implementation is sufficient).

However, if you want to keep the arn as a parameter, move it inside the properties, so that we can deprecate it in the future when MSK higher level construct is ready.

Reply:
There is a pending PR that defines [`msk.ICluster`](https://github.com/CurtisEppel/aws-cdk/blob/msk/l2-construct/packages/%40aws-cdk/aws-msk/lib/cluster.ts#L16) already. 
I'd move everything into props now and then add `ICluster` as first level constructor argument once it's available?

Reply:
I believe this is the PR you're referring to - https://github.com/aws/aws-cdk/pull/9908

If that gets merged in the next few days, we should go straight to the correct implementation. This would be preferred approach.

However, if the other PR is lagging, you could mark `ManagedKafkaEventSource` and `ManagedKafkaEventSourceProps` as `@experimental` and come back to change it later and remove the `@experimental` annotation.